### PR TITLE
[PREG-226] Use structs for conductor messages

### PR DESCRIPTION
**Pull request by jvolmer**
Use structs instead of plain velocypack for messages sent from conductor to workers. A simliar PR for messages from workers to conductor will follow.

This is a preparation step for using a proper messaging framework for Pregel.

### CODE REVIEW

**Comment by jvolmer** [comment]
TODO: merge parent first

File: arangod/Pregel/AlgoRegistry.cpp [hunk 1]
```diff
@@ -95,82 +96,76 @@ IAlgorithm* AlgoRegistry::createAlgorithm(
 
 template<typename V, typename E, typename M>
 /*static*/ std::shared_ptr<IWorker> AlgoRegistry::createWorker(
```
Comment (markuspf) [inline arangod/Pregel/AlgoRegistry.cpp:98]: shoudl probably just remove these `/* static */`

File: arangod/Pregel/Conductor/Conductor.cpp [hunk 2]
```diff
@@ -485,7 +493,7 @@ ErrorCode Conductor::_initializeWorkers(std::string const& suffix,
   std::string const path = Utils::baseUrl(Utils::workerPrefix) + suffix;
 
   // int64_t vertexCount = 0, edgeCount = 0;
```
Comment (markuspf) [inline arangod/Pregel/Conductor/Conductor.cpp:495]: Remove unused code.

File: arangod/Pregel/AlgoRegistry.cpp [hunk 3]
```diff
@@ -95,82 +96,76 @@ IAlgorithm* AlgoRegistry::createAlgorithm(
...
-    TRI_vocbase_t& vocbase, Algorithm<V, E, M>* algo, VPackSlice body,
-    PregelFeature& feature) {
-  return std::make_shared<Worker<V, E, M>>(vocbase, algo, body, feature);
+    TRI_vocbase_t& vocbase, Algorithm<V, E, M>* algo,
+    CreateWorker const& parameters, PregelFeature& feature) {
+  return std::make_shared<Worker<V, E, M>>(vocbase, algo, parameters, feature);
 }
 
 /*static*/ std::shared_ptr<IWorker> AlgoRegistry::createWorker(
```
Comment (markuspf) [inline arangod/Pregel/AlgoRegistry.cpp:104]: Same here

**Review by markuspf — APPROVED** [review]
Just some minor requests to remove commented out code

**Pull request merged**